ladybug/src/main/java/org/frankframework/ibistesttool/filter/GrayBox.java

```diff
@@ -20,20 +20,22 @@
 
 import nl.nn.testtool.Checkpoint;
 import nl.nn.testtool.Report;
-import nl.nn.testtool.filter.CheckpointMatcher;
 
 /**
  * Only show senders and pipelines (within these senders) (show only the pipeline checkpoints, not it's children)
  *
  * @author Jaco de Groot
  */
-public class GrayBox implements CheckpointMatcher {
-
+public class GrayBox extends AbstractBox {
+	@Override
 	public boolean match(Report report, Checkpoint checkpoint) {
 		if (checkpoint.getType() == Checkpoint.TYPE_INPUTPOINT || checkpoint.getType() == Checkpoint.TYPE_OUTPUTPOINT
 				|| checkpoint.getType() == Checkpoint.TYPE_INFOPOINT) {
 			List<Checkpoint> checkpoints = report.getCheckpoints();
 			ListIterator<Checkpoint> iterator = report.getCheckpoints().listIterator(checkpoints.indexOf(checkpoint));
+			if (hasStartPointOnLevel(report, checkpoint)) {
```

Comment:
This if can be moved two lines up so it is clear that this if doesn't use checkpoints and iterator.
Or maybe pass checkpoints and iterator to hasStartPointOnLevel() so that method doesn't have to reinvent the wheel but then we have to be sure the iterator is returned at the right position.

Reply:
I moved the if-statement up and changed the title of the PR